python/sglang/srt/torch_memory_saver_adapter.py

```diff
@@ -1,90 +1,118 @@
+        """
+        Initialize adapter with enable/disable control
 
-    def region(self):
-        return _primary_memory_saver.region()
+        Args:
+            enabled: Whether to enable memory saving functionality
+        """
+        self._user_enabled = enabled
 
-    def pause(self):
-        return _primary_memory_saver.pause()
+        if import_error is not None:
+            logger.warning(
+                "torch-memory-saver is not installed. Please install it "
+                "via `pip3 install torch-memory-saver`. "
+            )
 
-    def resume(self):
-        return _primary_memory_saver.resume()
+        logger.info(f"TorchMemorySaver adapter initialized with enabled={enabled}")
+
+    def region(self, tag: str):
+        """Context manager for memory region with specified tag"""
+        if self.enabled:
+            # Use the real torch_memory_saver context manager
+            logger.info(f"enter tms region for tag: {tag}")
+            return _memory_saver.region(tag=tag)
+        else:
+            # Use noop context manager when disabled
+            logger.debug(f"memory saver disabled, using noop context for tag: {tag}")
+            return _noop_context()
+
+    def pause(self, tag: str):
+        """Pause memory for specific tag"""
+        if self.enabled:
+            logger.info(f"enter tms pause for tag: {tag}")
+            _memory_saver.pause(tag=tag)
+        else:
+            logger.debug(f"memory saver disabled, noop pause for tag: {tag}")
+
+    def resume(self, tag: str):
+        """Resume memory for specific tag"""
+        if self.enabled:
+            logger.info(f"enter tms resume for tag: {tag}")
+            _memory_saver.resume(tag=tag)
+        else:
+            logger.debug(f"memory saver disabled, noop resume for tag: {tag}")
+
+    def pause_weights(self):
+        """Convenience method to pause weights memory"""
+        self.pause("weights")
+
+    def resume_weights(self):
+        """Convenience method to resume weights memory"""
+        self.resume("weights")
+
+    def pause_kv_cache(self):
+        """Convenience method to pause kv_cache memory"""
+        self.pause("kv_cache")
+
+    def resume_kv_cache(self):
+        """Convenience method to resume kv_cache memory"""
+        self.resume("kv_cache")
 
     @property
     def enabled(self):
-        return _primary_memory_saver.enabled
-
-
-class _TorchMemorySaverAdapterNoop(TorchMemorySaverAdapter):
-    @contextmanager
-    def configure_subprocess(self):
-        yield
-
-    @contextmanager
-    def region(self):
-        yield
-
-    def pause(self):
-        pass
+        """Check if memory saver is enabled (both user setting and library availability)"""
+        return (
+            self._user_enabled and _memory_saver is not None and _memory_saver.enabled
+        )
 
-    def resume(self):
-        pass
 
-    @property
-    def enabled(self):
-        return False
+# Factory function for backward compatibility
```

Comment:
nit: seems no need for backward compatiblity, since this is just an internal file